AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -114,80 +110,71 @@ CAmSocketHandler::~CAmSocketHandler()
 void CAmSocketHandler::start_listenting()
 {
     mDispatchDone = false;
-    int16_t pollStatus;
 
 #ifndef WITH_TIMERFD 
     clock_gettime(CLOCK_MONOTONIC, &mStartTime);
 #endif    
     timespec buffertime;
     
-    VectorListPoll_t cloneListPoll;
-    VectorListPoll_t::iterator listmPollIt;
-    VectorListPollfd_t fdPollingArray; //!<the polling array for ppoll
-    
-    auto preparePollfd = [&](const sh_poll_s& row)
-    {
-        CAmSocketHandler::prepare((sh_poll_s&)row);
-        pollfd temp = row.pollfdValue;
-        temp.revents = 0;
-        fdPollingArray.push_back(temp);
-    };
+    VectorPollfd_t fdPollingArray; //!<the polling array for ppoll
 
     while (!mDispatchDone)
     {
-        if (mRecreatePollfds)
+        // iterate all times though list and update ppoll accordingly
+        auto fdPollIt = fdPollingArray.begin();
+        for (auto it = mMapShPoll.begin(); it != mMapShPoll.end(); )
         {
-            fdPollingArray.clear();
-            //freeze mListPoll by copying it - otherwise we get problems when we want to manipulate it during the next lines
-            cloneListPoll = mListPoll;
-            //there was a change in the setup, so we need to recreate the fdarray from the list
-            std::for_each(cloneListPoll.begin(), cloneListPoll.end(), preparePollfd);
-            mRecreatePollfds = false;
-        }
-        else
-        {
-            //first we go through the registered filedescriptors and check if someone needs preparation:
-            std::for_each(cloneListPoll.begin(), cloneListPoll.end(), CAmSocketHandler::prepare);
+            auto& elem = it->second;
+            switch (elem.state)
+            {
+                case poll_states_e::REMOVE:
+                    close(elem.pollfdValue.fd);
+                    fdPollIt = fdPollingArray.erase(fdPollIt);
+                    it = mMapShPoll.erase(it);
+                    continue;
```

Comment:
I think I would have preferred a break over continue (and then of course put the increment of the iterators inside of the other case)

That's more following a pattern
if (condition)
   erase and assign new iterator
else
   increment iterator

But OK... it's not too bad.

Reply:
Actually your suggestion implies more "lines of code". Considering and if/elseif means that most of the cases I have to increment the iterator. This is what I wanted to avoid.

Reply:
Two more fall-throughs, from UPDATE, to VALID and finally fall through to default:, and then increment the iterator in default: would be the shortest.  But I might have missed a case that makes that not work...  In any case, it is probably not easier to understand, so this is fine.

Reply:
Yes, it will work. I like the idea. I consider to change also the order of the enum to align it better to the internal state machine:
```
typedef enum:uint8_t
{
    ADD     = 1u,
    UPDATE  = 2u,
    VALID   = 3u,
    CLOSE   = 4u,
    REMOVE  = 5u
} poll_states_e;
```